sklearn/preprocessing/_encoders.py

```diff
@@ -775,6 +775,26 @@ def fit(self, X, y=None):
                                      f"values already used for encoding the "
                                      f"seen categories.")
 
+        # Check for None values in categories
+        for cat_idx, categories_for_idx in enumerate(self.categories_):
+            # None and np.nan will only appear at the end of feature_cats
+            if None in categories_for_idx[-2:]:
+                raise ValueError(
+                    "None is an unsupported missing value indicator and was "
```

Comment:
FWIW, this is backwards incompatible. We currently transform None as another category label. Are we sure we want to do this?

Reply:
Ah I did not know `None` was already supported. This kind of changes things a little.

Reply:
`None` support doesn't seem to be tested nor documented anywhere though. If we choose to keep the previous behaviour we probably want to add tests and docs